frontend/individual_pedigree.tsx

```diff
@@ -0,0 +1,100 @@
+
+import React, { Component, useEffect } from 'react'
+import { Link } from "react-router-dom";
+import { get } from './communication';
+import { PedigreeNetwork } from "./pedigree"
+import Table from '@material-ui/core/Table';
+import TableBody from '@material-ui/core/TableBody';
+import TableCell from '@material-ui/core/TableCell';
+import TableRow from '@material-ui/core/TableRow';
+
+
+/**
+ * Shows the information of a given individual and the pedigree graph built using the PedigreeNetwork component
+ */
+export function IndividualPedigree({ id, generations }: { id: string }) {
+  const [pedigree, setPedigree] = React.useState(undefined as any)
+  const [individual, setIndividual] = React.useState(undefined as any)
+  const [generations_input, setGenerations] = React.useState(generations)
+
+
+  React.useEffect(() => {
+
+    let mounted = true
+    const url = `/api/individual/${id}`
+    get(url).then(
+      data => {
+        mounted && data && setIndividual(data)
+      },
+      error => console.error(error)
+    )
+    get(`/api/pedigree/${id}/${generations_input}`).then(
+      data => mounted && data && setPedigree(data),
+      error => console.error(error)
+    )
+
+    return () => { mounted = false }//allows to validate the mount state of the component
+
+  }, [id, generations_input])//dependent on these values
+
+  return <>
+    {individual && pedigree && <>//the component will be rendered if the individual and pedigree were retrieved
```

Comment:
this comment is rendered on the page :) it needs to be wrapped up as `{// comment}`

Reply:
upss

Reply:
solved